pkg/nuctl/command/common/helpers.go

```diff
@@ -111,6 +112,23 @@ func saveReportToFile(ctx context.Context, loggerInstance logger.Logger, report
 			"err", err,
 			"path", path)
 	}
+	// get the directory path from the file path
+	dir := filepath.Dir(path)
+
+	// check if the directory exists, create it if it doesn't
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		loggerInstance.DebugWithCtx(ctx,
+			"Creating directory for report as it does not exist",
+			"directory", dir)
+
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			loggerInstance.ErrorWithCtx(ctx,
+				"Failed to create directory for report",
+				"directory", dir,
+				"error", err)
+		}
+	}
```

Comment:
This is a classic helper function `EnsureDirExists`.
move to `pkg/common/helper.go` (just remember to change the logs to not have `report` in them).

Reply:
@TomerShor I'm uncertain if testing the library is necessary, but I've included a simple test to demonstrate that the length is zero despite the presence of headers.